runtime/src/stake_delegations.rs

```diff
@@ -0,0 +1,86 @@
+//! Map pubkeys to stake delegations
+//!
+//! This module implements clone-on-write semantics for `StakeDelegations` to reduce unnecessary
+//! cloning of the underlying map.
+use {
+    solana_sdk::{pubkey::Pubkey, stake::state::Delegation},
+    std::{
+        collections::HashMap,
+        ops::{Deref, DerefMut},
+        sync::Arc,
+    },
+};
+
+/// A map of pubkey-to-stake-delegation with clone-on-write semantics
+#[derive(Default, Clone, PartialEq, Debug, Deserialize, Serialize)]
+pub struct StakeDelegations(Arc<StakeDelegationsInner>);
```

Comment:
The inner map needs to be wrapped in an `Arc` due to it being passed between threads. Found this out from the compiler while building an earlier version with `Rc`.

jstarry Using `Rc` (or `Arc`) answers the questions we were having yesterday about Rust and lifetimes, vs just using `Cow`.